src/core/meshcop/border_agent.cpp

```diff
@@ -317,6 +333,187 @@ template <> void BorderAgent::HandleTmf<kUriRelayRx>(Coap::Message &aMessage, co
     FreeMessageOnError(message, error);
 }
 
+void BorderAgent::NotifyMeshCoPServiceChange(void)
+{
+    MeshCoPTxtEncoder meshCoPTxtEncoder(GetInstance());
+
+    VerifyOrExit(mMeshCoPServiceChangedCallback.IsSet());
+    OT_ASSERT(meshCoPTxtEncoder.EncodeTxtData() == kErrorNone);
+
+    mMeshCoPServiceChangedCallback.Invoke(meshCoPTxtEncoder.GetTxtData(), meshCoPTxtEncoder.GetTxtDataLen());
+
+exit:
+    return;
+}
+
+void BorderAgent::PostNotifyMeshCoPServiceChangeTask(void)
+{
+    if (mMeshCoPServiceChangedCallback.IsSet())
+    {
+        mNotifyMeshCoPServiceChangeTask.Post();
+    }
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+// BorderAgent::MeshCoPTxtEncoder
+
+Error BorderAgent::MeshCoPTxtEncoder::AppendTxtEntryTo(const char *aKey, const void *aValue, uint16_t aValueLength)
```

Comment:
Sorry for many smaller suggestion. 

How about we rename to `AppendTxEntry()` (removing the `To`). Since now it does not get `Appender` or `Message` as input anymore?

Reply:
More than happy to get so many nice suggestions :) Thanks!